tests/integration/test_examples.py

```diff
@@ -141,6 +142,41 @@ async def test_task_pool(flow, scheduler, one_conf):
     assert len(schd.pool.pool) == 1
 
 
+@pytest.mark.asyncio
+async def test_mock_task_pool(flow, scheduler, run):
+    """Any you don't have to run tasks to get the pool in the right state."""
+    # Ensure Cylc can load tasks from the DB on restart
+
+    # create a flow
+    reg = flow({
+        'scheduling': {
+            'graph': {
+                'R1': 'a & b & c'
+            }
+        }
+    })
+
+    # initiate it as a restart with some jobs in pre-defined states
+    tasks = [
+        Task('a', '1', 'waiting'),
+        Task('b', '1', 'running'),
+        Task('c', '1', 'failed'),
+    ]
+    schd = scheduler(reg, tasks=tasks)
```

Comment:
From a test perspective it's pretty straight forward, just provide a list of `Task` instances to the `scheduler` fixture.